### DISCUSSION TITLE
Backup local(K8S) and remote(OBS) state consistency processing needs to be enhanced.

### DISCUSSION THREAD

**Discussion by neujie** [vmware-tanzu/velero#3268] [Community support Q&A] [answered]
**Describe the problem/challenge you have**
[A description of the current limitation/problem/challenge that you are experiencing.]

Currently velero processes Backup as follows:  
1. patch backup's status `BackupPhaseInProgress` to K8S before invoke `runBackup()`;
2. `runBackup()`:  
2.1 backup K8S resource and invoke some plugins to backup volumes  
2.2 set backup status, like `BackupPhasePartiallyFailed`、`BackupPhaseCompleted` 、`BackupPhaseFailed`, not patch to K8S;  
2.3 invoke `persistBackup()`   
3. patch the backup's status to K8S;

So, some abnormal scenario were appeared：  
1. Velero quit while making a backup, then the backup's status will be frozen(abandoned);
2. Velero exited while uploading the backup files, then The local state(K8S) and remote state(OBS) are inconsistent;

Like #3144, when velero was OOM in K8S, the backup was broken off and abandoned.   
So, the backup's status still keep `InProcessing` all time.
`backup_controller` and `backup_sync_controller` can't deal these `InProcessing` backups.

**The lifecycle management of backup should be enhanced in velero.** 

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

Maybe we can：  
1. Ensure the consistency of backup operations;
2. Supports interrupt continuation capabilities;
3. Supports intermittent transmission of data.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Velero version (use `velero version`):
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

**Comment by dsu-igeek** [comment, marked as answer]
We'll be improving this area as part of https://github.com/vmware-tanzu/velero/issues/3533.
